pkg/securitypolicy/securitypolicyenforcer_rego.go

```diff
@@ -21,50 +36,491 @@ func init() {
+		values[i] = mount.MarshalRego()
+	}
+
+	_, err := builder.WriteString(fmt.Sprintf("%s\"mounts\": [%s],\n", indent, strings.Join(values, ",")))
+	return err
+}
+
+func writeContainer(builder *strings.Builder, container *securityPolicyContainer, indent string) error {
+	if _, err := builder.WriteString(fmt.Sprintf("%s{\n", indent)); err != nil {
+		return err
+	}
+
+	if err := writeCommand(builder, container.Command, indent+Indent); err != nil {
+		return err
+	}
+
+	if err := writeEnvRules(builder, container.EnvRules, indent+Indent); err != nil {
+		return err
+	}
+
+	if err := writeLayers(builder, container.Layers, indent+Indent); err != nil {
+		return err
+	}
+
+	if err := writeMounts(builder, container.Mounts, indent+Indent); err != nil {
+		return err
+	}
+
+	if _, err := builder.WriteString(fmt.Sprintf("%s\"allow_elevated\": %v,\n", indent+Indent, container.AllowElevated)); err != nil {
+		return err
+	}
+
+	if _, err := builder.WriteString(fmt.Sprintf("%s\"working_dir\": \"%s\"\n", indent+Indent, container.WorkingDir)); err != nil {
+		return err
+	}
+
+	if _, err := builder.WriteString(fmt.Sprintf("%s}", indent)); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func addContainers(builder *strings.Builder, containers []*securityPolicyContainer) error {
+	if _, err := builder.WriteString("containers := [\n"); err != nil {
+		return err
+	}
+
+	for i, container := range containers {
+		if err := writeContainer(builder, container, Indent); err != nil {
+			return err
+		}
+
+		var end string
+		if i < len(containers)-1 {
+			end = ",\n"
+		} else {
+			end = "\n"
+		}
+
+		if _, err := builder.WriteString(end); err != nil {
+			return err
+		}
+	}
+
+	if _, err := builder.WriteString("]\n"); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (p securityPolicyInternal) MarshalRego() (string, error) {
+	builder := new(strings.Builder)
+	if _, err := builder.WriteString(fmt.Sprintf("package policy\nallow_all := %v\n", p.AllowAll)); err != nil {
+		return "", err
+	}
+
+	if err := addContainers(builder, p.Containers); err != nil {
+		return "", err
+	}
+
+	return builder.String(), nil
+}
+
+func createRegoEnforcer(state SecurityPolicyState, defaultMounts []oci.Mount, privilegedMounts []oci.Mount) (SecurityPolicyEnforcer, error) {
+	policy, err := state.SecurityPolicy.toInternal()
+	if err != nil {
+		return nil, fmt.Errorf("error converting to internal format: %w", err)
+	}
+
+	regoPolicy, err := newRegoPolicyFromInternal(policy, defaultMounts, privilegedMounts)
+	if err != nil {
+		return nil, fmt.Errorf("error converting to Rego: %w", err)
+	}
+
+	regoPolicy.base64policy = state.EncodedSecurityPolicy.SecurityPolicy
+	return regoPolicy, nil
+}
+
+func newRegoPolicyFromInternal(securityPolicy *securityPolicyInternal, defaultMounts []oci.Mount, privilegedMounts []oci.Mount) (*RegoEnforcer, error) {
+	policy := new(RegoEnforcer)
+	policy.behavior = PolicyCode
+	if code, err := securityPolicy.MarshalRego(); err == nil {
```

Comment:
Now that we are using Rego, can we switch to having the user pass us a Rego document directly? Then we don't have to do any parsing, and can just pass it to the Rego compiler directly.

I guess we would still want a JSON -> Rego path for existing users who pass JSON already, but it seems like the happy path should support passing Rego. Thoughts?

Reply:
@anmaxvl you might have opinions on this?

Reply:
The work to support Rego directly is coming in subsequent PRs. As an initial integration, we decided to support the current JSON policy and internally convert it to Rego for evaluation. This enabled us to start testing Rego engine without fully implementing tooling around it, since we have integration tests for JSON policy.

Reply:
Okay, so the current approach of JSON -> internal representation -> Rego is temporary? I'm really looking forward to moving away from the current approach and legacy JSON in the future and just taking a Rego directly :D

Reply:
That's the next PR.

Reply:
@kevpar , yes 😄 it's all temporary... as @SeanTAllen mentioned, direct Rego ingestion is coming next